### PR TITLE
Story one

### PR DESCRIPTION
**Pull request by marcusxavier85**
This is the one

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Links the PHP to the correct Database
+ *
+ * @param none
+ *
+ * @return PDO, returning link to database
+ */
+function getDatabaseObject(): PDO {
+
+    return $db = new PDO('mysql:host=db;dbname=pokedex', 'root', 'password');
```
Comment (mporam) [inline functions.php]: return means the rest of the function will not execute, so the set attribute may not work. you should return `$db` after the set attribute

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+function getPokemons(PDO $db): array {
+    $query = $db->prepare('SELECT `id`, `pokedex_no`, `name`, `type`, `img_source` FROM `kanto_pokedex`;');
+
+    $query->execute();
+
+    return $query->fetchAll();
+}
+
+/**
+ * Displays data gathered in the table pn the browser
```
Comment (mporam) [inline functions.php]: typo 'pn'

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * @param $pokemons, nested associative array, used as the base to grab the data from
+ *
+ * @return string, returning a concatination of all data in one array and then doing it again for the other nested arrays
+ */
+function displayPokemon(array $pokemons): string {
+    $result = '';
+    foreach ($pokemons as $pokemon) {
+        if (array_key_exists('img_source', $pokemon) && array_key_exists('pokedex_no', $pokemon) && array_key_exists('name', $pokemon) && array_key_exists('type', $pokemon)) {
```
Comment (mporam) [inline functions.php]: long if statements should be laid out:
```
if (
    condition &&
    condition &&
    condition
) {
```

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            $result .= '<tr class="rowBox">';
+            $result .= '<td><img class="pictureOfPokemon" src=' . $pokemon['img_source'] . ' alt="Picture of "' . $pokemon['name'] . '</td>';
+            $result .= '<td>' . $pokemon['pokedex_no'] . '</td>';
+            $result .= '<td>' . $pokemon['name'] . '</td>';
+            $result .= '<td>' . $pokemon['type'] . '</td>';
+            $result .= '</tr>';
+        } else {
+            return 'This is not correct please reconsider';
```
Comment (mporam) [inline functions.php]: if you pass in an array where the first pokemon is correct, the second is missing a key, the third is correct, what will happen?

File: tests/functions.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    // if incorrect data type fails
+    public function testMalformedDisplayPokemons () {
+            $this->expectException(TypeError::class);
+            $input = 89;
+            $result = displayPokemon($input);
+
+    }
+
+    // a few failure tests checking each key
```
Comment (mporam) [inline tests/functions.php]: remove comments

**Review by mporam — CHANGES_REQUESTED** [review]

File: functions.php (hunk 1 above)
Comment (marcusxavier85) [reply functions.php]: Amended

File: functions.php (hunk 2 above)
Comment (marcusxavier85) [reply functions.php]: Amended

File: functions.php (hunk 3 above)
Comment (marcusxavier85) [reply functions.php]: Amended

File: tests/functions.php (hunk 5 above)
Comment (marcusxavier85) [reply tests/functions.php]: Done

File: functions.php (hunk 4 above)
Comment (marcusxavier85) [reply functions.php]: Amedned

File: functions.php [hunk 6]
```diff
@@ -39,15 +41,16 @@ function getPokemons(PDO $db): array {
 function displayPokemon(array $pokemons): string {
     $result = '';
     foreach ($pokemons as $pokemon) {
-        if (array_key_exists('img_source', $pokemon) && array_key_exists('pokedex_no', $pokemon) && array_key_exists('name', $pokemon) && array_key_exists('type', $pokemon)) {
+        if (array_key_exists('img_source', $pokemon) &&
```
Comment (mporam) [inline functions.php]: still not correct, check my last comment. first and last condition should be on their own line

**Review by mporam — CHANGES_REQUESTED** [review]

File: functions.php (hunk 6 above)
Comment (marcusxavier85) [reply functions.php]: Sorry didn't spot that, all sorted.

File: pokedexForm.php [hunk 7]
```diff
@@ -2,9 +2,9 @@
     <body>
```
Comment (mporam) [inline pokedexForm.php]: this file is nothing to do with story 1, please delete

**Review by mporam — CHANGES_REQUESTED** [review]

File: pokedexForm.php (hunk 7 above)
Comment (marcusxavier85) [reply pokedexForm.php]: Have removed said file plus some others which weren't needed

**Review by mporam — APPROVED** [review]

**Pull request merged**